test/e2e/operator_test.go

```diff
@@ -758,6 +759,118 @@ func TestScopeChange(t *testing.T) {
+		t.Skipf("test skipped on platform %q", platform)
+	}
+
+	t.Log("Creating ingresscontroller with spec.endpointPublishingStrategy.loadBalancer.allowedSourceRanges set to 127.0.0.0/8")
+	name := types.NamespacedName{Namespace: operatorNamespace, Name: "sourcerange"}
+	domain := name.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newLoadBalancerController(name, domain)
+	ic.Spec.EndpointPublishingStrategy.LoadBalancer = &operatorv1.LoadBalancerStrategy{
+		Scope:               operatorv1.ExternalLoadBalancer,
+		AllowedSourceRanges: []operatorv1.CIDR{"127.0.0.0/8"},
+	}
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	// Wait for the load balancer and DNS to be ready.
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, name, nondefaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	lbService := &corev1.Service{}
+	if err := kclient.Get(context.TODO(), controller.LoadBalancerServiceName(ic), lbService); err != nil {
+		t.Fatalf("failed to get service: %v", err)
+	}
+
+	if isServiceInternal(lbService) {
+		t.Fatalf("load balancer is internal but should be external")
+	}
+
+	n := 1
+	checkRoute := func() (string, error) {
+		// Use a unique host name for each request to work around
+		// caching of negative responses.
+		url := fmt.Sprintf("http://x%d.%s./", n, domain)
+		n++
+
+		request, err := http.NewRequest("GET", url, nil)
+		if err != nil {
+			return "", err
+		}
+
+		client := &http.Client{Timeout: 5 * time.Second}
+		response, err := client.Do(request)
+		if err != nil {
+			return "", err
+		}
+
+		_, err = io.Copy(ioutil.Discard, response.Body)
+		response.Body.Close()
+		if err != nil {
+			return "", err
+		}
+
+		return response.Status, nil
+	}
+
+	// The initial source range of 127.0.0.0/8 should block all traffic.
+	// Poll for a minute and verify that we do not get an HTTP response.
+	gotResponse := false
+	err := wait.PollImmediate(10*time.Second, 1*time.Minute, func() (bool, error) {
+		status, err := checkRoute()
+		if err != nil {
+			t.Logf("Got error from LB: %v", err)
+			return false, nil
+		}
+		t.Logf("Got HTTP response from LB: %s", status)
+		gotResponse = true
+		return true, nil
+	})
+	if gotResponse {
+		t.Fatal("expected load balancer to be inaccessible")
+	}
+
+	// Remove the allowed source range so that the default of 0.0.0.0/0
+	// (that is, allow all) takes effect.
+	t.Log("Clearing allowed source ranges")
+	if err := kclient.Get(context.TODO(), name, ic); err != nil {
+		t.Fatalf("failed to get ingresscontroller %s: %v", name, err)
+	}
+	ic.Spec.EndpointPublishingStrategy.LoadBalancer.AllowedSourceRanges = []operatorv1.CIDR{}
+	if err := kclient.Update(context.TODO(), ic); err != nil {
+		t.Fatal(err)
+	}
+
+	err = wait.PollImmediate(10*time.Second, 2*time.Minute, func() (bool, error) {
```

Comment:
Any particular reason that this loop has a timeout of 2 minutes, but the prior loop has a timeout of only 1 minute?

Reply:
I have a couple reasons, but I'm not sure how sound they are.  First, the first timeout is _expected_ to be hit (the request should always fail), so it's a question of how long we want the test to sit around making sure that the thing that should never happen never happens.  One minute seems like a reasonable amount of time to be confident that the LB really isn't accepting connections, without increasing the overall test time in the expected case too much.  Second, the second timeout is expected _not_ to be hit, and if we only hit it because the configuration change takes longer than expected, that's a test flake, so the high timeout usually won't increase the overall test time, and when it does, it reduces flakiness, which is worth waiting an extra minute (maybe even a couple extra minutes).  Does that sound reasonable?

Reply:
> Does that sound reasonable?

Yup. We can always readjust these values in the future need be.